python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/dlt/resource.py

```diff
@@ -112,28 +115,32 @@ def run(
                 META_KEY_PIPELINE,
             )
 
-        # Mapping of asset keys to dlt resources
-        asset_key_dlt_source_resource_mapping = zip(
-            context.selected_asset_keys, dlt_source.resources.values()
-        )
+        asset_key_dlt_source_resource_mapping = {
```

Comment:
This will be the same for every asset, so let's fetch alongside our other named metadata.

```suggestion
        dlt_source_resource = first_asset_metadata.get(META_KEY_RESOURCE)
```

Reply:
I don't believe it will; each asset is tied to a specific _dlt_ resource. I think the confusion was _dlt_ resource vs. Dagster resource. Stupid name conflicts.
```python
        specs = [
            AssetSpec(
                key=dlt_dagster_translator.get_asset_key(dlt_source_resource),
                deps=dlt_dagster_translator.get_deps_asset_keys(dlt_source_resource),
                auto_materialize_policy=dlt_dagster_translator.get_auto_materialize_policy(
                    dlt_source_resource
                ),
                metadata={  # type: ignore - source / translator are included in metadata for use in `.run()`
                    META_KEY_SOURCE: dlt_source,
                    META_KEY_PIPELINE: dlt_pipeline,
                    META_KEY_RESOURCE: dlt_source_resource,
                },
            )
            for dlt_source_resource in dlt_source.resources.values()
        ]
```

Reply:
Oh whoops, you're right. I didn't realize `dlt_source_resource` wasn't an argument.

A nit then would be to have a more explicit name (since `XXX_resource` is flying around in multiple places). Maybe `dlt_resource_for_source`